ui/src/state/diary/diary.ts

```diff
@@ -535,49 +306,919 @@ export function useBrief(flag: string) {
+            quippers: replyingNote.quippers.filter(
+              (quipper) => quipper !== window.our
+            ),
+          };
+
+          queryClient.setQueryData<Record<string, DiaryLetter>>(
+            ['diary', 'notes', variables.flag],
+            {
+              ...notes,
+              [replying]: updatedNote,
+            }
+          );
+        }
+      }
+    },
+  });
+}
+
+export function useAddNoteFeelMutation() {
+  const queryClient = useQueryClient();
+  const mutationFn = async (variables: {
+    flag: DiaryFlag;
+    noteId: string;
+    feel: string;
+  }) => {
+    const diff: DiaryDiff = {
+      notes: {
+        time: decToUd(variables.noteId),
+        delta: {
+          'add-feel': {
+            time: decToUd(unixToDa(Date.now()).toString()),
+            feel: variables.feel,
+            ship: window.our,
+          },
+        },
+      },
+    };
+
+    await api.poke(diaryAction(variables.flag, diff));
+  };
+
+  return useMutation({
+    mutationFn,
+    onMutate: async (variables) => {
+      await queryClient.cancelQueries([
+        'diary',
+        'notes',
+        variables.flag,
+        variables.noteId,
+      ]);
+
+      const prevNote = queryClient.getQueryData<DiaryNote>([
+        'diary',
+        'notes',
+        variables.flag,
+        variables.noteId,
+      ]);
+
+      if (prevNote) {
+        const prevFeels = prevNote.seal.feels;
+        const newFeels = {
+          ...prevFeels,
+          [unixToDa(Date.now()).toString()]: variables.feel,
+        };
+
+        const updatedNote: DiaryNote = {
+          ...prevNote,
+          seal: {
+            ...prevNote.seal,
+            feels: newFeels,
+          },
+        };
+
+        queryClient.setQueryData<DiaryNote>(
+          ['diary', 'notes', variables.flag, variables.noteId],
+          updatedNote
+        );
+      }
+    },
+    onSettled: async (_data, _error, variables) => {
+      await queryClient.refetchQueries(['diary', 'notes', variables.flag]);
+      await queryClient.refetchQueries([
+        'diary',
+        'notes',
+        variables.flag,
+        variables.noteId,
+      ]);
+    },
+  });
+}
+
+export function useDeleteNoteFeelMutation() {
+  const queryClient = useQueryClient();
+  const mutationFn = async (variables: { flag: DiaryFlag; noteId: string }) => {
+    const diff: DiaryDiff = {
+      notes: {
+        time: decToUd(variables.noteId),
+        delta: {
+          'del-feel': window.our,
+        },
+      },
+    };
+
+    await api.poke(diaryAction(variables.flag, diff));
+  };
+
+  return useMutation({
+    mutationFn,
+    onMutate: async (variables) => {
+      await queryClient.cancelQueries([
+        'diary',
+        'notes',
+        variables.flag,
+        variables.noteId,
+      ]);
+
+      const prevNote = queryClient.getQueryData<DiaryNote>([
+        'diary',
+        'notes',
+        variables.flag,
+        variables.noteId,
+      ]);
+
+      if (prevNote) {
```

Comment:
we should probably wrap this up in a function since we're doing it over and over and just have it accept some update function that returns a new note